apps/condo/domains/meter/tasks/sendVerificationDateReminder.js

```diff
@@ -0,0 +1,245 @@
+                })
+                return message == null
+            })
+
+            // create a pair again
+            // and filter out meter with empty residents
+            return {
+                meter,
+                residents: residentsWithoutNotifications,
+            }
+        })
+        .filter(item => item.residents.length > 0)
+}
+
+const getOrganizationLang = async (context, id) => {
+    const organization = await Organization.getOne(context, { id, deletedAt: null })
+
+    /**
+     * Detect message language
+     * Use DEFAULT_LOCALE if organization.country is unknown
+     * (not defined within @condo/domains/common/constants/countries)
+     */
+    return get(COUNTRIES, get(organization, 'country.locale'), DEFAULT_LOCALE)
+}
+const generateReminderMessages = async ({ context, reminderWindowSize, reminders }) => {
+    const messages = []
+
+    await Promise.all(reminders.map(async (reminder) => {
+        const { meter, residents } = reminder
+        const lang = await getOrganizationLang(context, meter.organization.id)
+
+        // prepare a message for each resident
+        messages.push(
+            ...residents.map(resident => ({
+                sender: { dv: 1, fingerprint: 'meters-validation-date-cron-push' },
+                to: { user: { id: resident.user.id } },
+                type: METER_VERIFICATION_DATE_REMINDER_TYPE,
+                lang,
+                meta: {
+                    dv: 1,
+                    data: {
+                        reminderDate: dayjs(meter.nextVerificationDate).locale(lang).format('D MMM'),
+                        meterId: meter.id,
+                        userId: resident.user.id,
+                        residentId: resident.id,
+                    },
+                },
+            }))
+        )
+    }))
+
+    return messages
+}
+
+const sendReminders = async ({ context, messages }) => {
+    await Promise.all(messages.map(async (message) => {
+        await sendMessage(context, message)
+    }))
+}
+
+const sendVerificationDateReminder = async ({ date, searchWindowDaysShift, daysCount }) => {
+    // prepare vars
+    if (!date) date = dayjs().format('YYYY-MM-DD')
+    const reminderWindowSize = searchWindowDaysShift + daysCount
+
+    // initialize context stuff
+    const { keystone: context } = await getSchemaCtx('Meter')
+
+    // let's proceed meters page by page
+    const pageSize = 100
+    let offset = 0
+    let hasMore = false
+    do {
+        const metersToCheck = await readMetersPage({
+            context, offset, pageSize, date, searchWindowDaysShift, daysCount,
+        })
+
+        // remove meters with wrong nextVerificationDate
+        const meters = metersToCheck.filter(({ verificationDate, nextVerificationDate }) => {
+            return nextVerificationDate && verificationDate && dayjs(nextVerificationDate).diff(verificationDate, 'month') > 1
+        })
```

Comment:
can we just add this filter condition inside the `readMetersPage`

Reply:
Nope. We need to have metersToCheck and meters (filtered one).  metersToCheck - to control pagination (hasMore = metersToCheck.length > 0)